homework/grayscale-image/compression.cpp

```diff
@@ -0,0 +1,38 @@
+#include "compression.hpp"
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& pgmImage) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressedPGM;
+    for (auto& pixelRow : pgmImage) {
+        uint8_t numberOfOccurences{1};
+        uint8_t pixelGreyScale{0};
+        std::array<uint8_t, width>::const_iterator it;
+        for (it = pixelRow.begin(); it < pixelRow.end() - 1; it++) {
```

Comment:
If you refactor this loop, you don't need to declare it outside loop.

for (auto it = begin() ....) 
See note below